### PR TITLE
added support for private keys in the ftp component

### PR DESCRIPTION
**Pull request by lucvankessel**
We were in need of a key option in the FTP component.
This is the implementation i came up with.
I have tested it  that it still works with password and the new key option.
Let me know if there are things you would like to see different or changed.

### CODE REVIEW

File: src/appmixer/utils/ftp/client.js [hunk 1]
```diff
@@ -79,7 +85,23 @@ class FtpClient {
      */
     static isFtp(secure) {
 
-        return ['yes', 'implicit', ''].includes(secure);
+        return ['yes', 'implicit'].includes(secure);
```
Comment (lucvankessel) [inline src/appmixer/utils/ftp/client.js:88]: the `''` the default to be the secure option

**Comment by jirihofman** [comment]
Thank you for your contribution @lucvankessel 🎉 . We will review the change some time next week, even though it looks good at first glance.

File: src/appmixer/utils/ftp/client.js [hunk 2]
```diff
@@ -14,10 +15,15 @@ class FtpClient {
     constructor(secure, config) {
 
         this.config = config;
-        this.isFtp = FtpClient.isFtp(secure);
-        if (this.isFtp) {
+
+        if (FtpClient.isFtp(secure)) {
+            if (config.privateKey != '') {
+                throw new Error('Cant use the secure option in combination with a private key');
```
Comment (jirihofman) [inline src/appmixer/utils/ftp/client.js:21]:
```suggestion
                throw new Error('Private key can be used only for SFTP');
```

File: src/appmixer/utils/ftp/auth.js [hunk 3]
```diff
@@ -29,6 +28,11 @@ module.exports = {
                 name: 'Password',
                 tooltip: 'Password.'
             },
+            privatekey: {
+                type: 'textarea',
+                name: 'PrivateKey',
+                tooltip: 'Private key; either use this or the password field, this only works with sFTP. PEM is the only supported format for keys that can be used.'
```
Comment (jirihofman) [inline src/appmixer/utils/ftp/auth.js:34]:
```suggestion
                tooltip: 'Private key; either use this or the password field. This only works with SFTP. PEM is the only supported format for keys that can be used.'
```

**Review by jirihofman — CHANGES_REQUESTED** [review]
Minor wording update

File: src/appmixer/utils/ftp/auth.js (hunk 3 above)
Comment (lucvankessel) [reply src/appmixer/utils/ftp/auth.js]: @jirihofman implemeted both changes

**Review by jirihofman — APPROVED** [review]
Great, thank you @lucvankessel

I just bumped the version and added changelog entry

**Pull request merged**